### PR TITLE
Migrate from BiocParallel to future

### PR DESCRIPTION
**Pull request by lwaldron**
This PR replaces the BiocParallel package with the future framework and future.apply. The legacy BPPARAM argument has been deprecated and now triggers a deprecation warning while gracefully ignoring it via dot arguments.

### CODE REVIEW

File: tests/testthat/test_DoppelGang-class.R [hunk 1]
```diff
@@ -20,9 +20,7 @@ n.eset$id <- toupper(colnames(n1))
 esets <- list(m = m.eset, n = n.eset)
 
 test_that("DoppelGang class methods work (Normal Use)", {
-  suppressWarnings(suppressMessages(
-    res <- doppelgangR(esets, BPPARAM = BiocParallel::SerialParam())
-  ))
+    res <- doppelgangR(esets)
```
[the remark this answers is not in the document]
Comment (lwaldron) [reply tests/testthat/test_DoppelGang-class.R:24]: Set `verbose = FALSE`

File: tests/testthat/test_DoppelGang-class.R [hunk 2]
```diff
@@ -52,9 +50,7 @@ test_that("DoppelGang class methods work (Normal Use)", {
 })
 
 test_that("DoppelGang plot method skip.no.doppels and plot.pair arguments work (Edge Cases)", {
-  suppressWarnings(suppressMessages(
-    res <- doppelgangR(esets, BPPARAM = BiocParallel::SerialParam())
-  ))
+    res <- doppelgangR(esets)
```
[the remark this answers is not in the document]
Comment (lwaldron) [reply tests/testthat/test_DoppelGang-class.R:54]: Set `verbose = FALSE`

File: tests/testthat/test_doppelgangR_edgecases.R [hunk 3]
```diff
@@ -67,6 +67,6 @@ test_that("doppelgangR handles edge cases and error states correctly", {
   # Let's just create an eset with non-numeric matrix to throw an error
   eset_fail <- ExpressionSet(assayData = matrix("A", ncol=2, nrow=2), phenoData = AnnotatedDataFrame(data.frame(age=1:2)))
   expect_error(
-    doppelgangR(list(eset1_sg, eset_fail), BPPARAM = BiocParallel::SerialParam(stop.on.error = FALSE))
+    doppelgangR(list(eset1_sg, eset_fail))
```
[the remark this answers is not in the document]
Comment (lwaldron) [reply tests/testthat/test_doppelgangR_edgecases.R]: Remove mention of bplapply
